### ISSUE TITLE
Supporting OpenAPI 3.1

### ISSUE THREAD

**Issue by OliverDudgeon** [anymaniax/orval#386]
Some progress seems to have (finally!) be made on `@apidevtools/swagger-parser`. There is now a release 10.1.0 that should allow OpenApi 3.1 to be parsed. Does this mean Orval could be updated to be compatible or is there still more that is needed with dependencies?

**Comment by anymaniax** [comment]
Hello @OliverDudgeon, Do you have time to do the update and test if everything works properly?

**Comment by OliverDudgeon** [comment]
An initial basic test seems to work for me. I just updated the version of `@apidevtools/swagger-parser` and ran it on [this 3.1 spec](https://paste.ee/p/4gK9S) and there were no ts errors. I'll try do some more testing if I get the time.

**Comment by OliverDudgeon** [comment]
Also the tests pass. I've tried it with a few different configs without issues. Seems like an easy upgrade? Maybe give it a few days for any reports to come through on @apidevtools/swagger-parser (it also has a PR that's been merged since the release so there might be another patch incoming)

**Comment by melloware** [comment]
Can this be closed since it was updated to 10.1.0 in master and ill be in the next release?

**Comment by anymaniax** [comment]
Released with 6.9.0

**Comment by geirsagberg** [comment]
Seems like the nullable part of 3.1 is not yet supported? From https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0 

> In line with JSON Schema, the `type` keyword can now define multiple types for a schema with an array. This is useful new functionality, but has also made nullable redundant. In order to achieve the goal of letting JSON Schema tools understand OpenAPI, it was decided to remove nullable entirely instead of deprecate it. 

```
# OpenAPI v3.0
type: string
nullable: true

# OpenAPI v3.1
type:
- "string"
- "null" 
```
